rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -172,9 +169,157 @@ NodeParameters::NodeParameters(
 NodeParameters::~NodeParameters()
 {}
 
+RCLCPP_LOCAL
+bool
+__lockless_has_parameter(
```

Comment:
@wjwwood out of curiosity, I see below that sometimes this function is used and sometimes its implementation is copied _almost_ verbatim. Why is that?

Reply:
I see a lot of cases where this is happening, but all are either of a different type for the first parameter or they explicitly store the iterator, compare it to end(), and then use the iterator, which this function does not allow you to do because the iterator is not returned.